Source/utils/soundsample.h

```diff
@@ -13,11 +16,20 @@ class SoundSample final {
 	void Play(int lVolume, int lPan, int channel = -1);
 	void Stop();
 	int SetChunkStream(HANDLE stormHandle);
-	int SetChunk(BYTE *fileData, DWORD dwBytes);
+
+	/**
+	 * @brief Sets the sample's WAV, FLAC, or Ogg/Vorbis data.
+	 * @param fileData Buffer containing the data
+	 * @param dwBytes Length of buffer
+	 * @return 0 on success, -1 otherwise
+	 */
+	int SetChunk(std::unique_ptr<std::byte[]> file_data, DWORD dwBytes);
+
 	int GetLength();
 
 private:
-	Mix_Chunk *chunk;
+	std::unique_ptr<std::byte[]> file_data_;
+	std::optional<Aulib::Stream> stream_;

```

Comment:
Not sure why you're using an optional instead of a unique_ptr here?

Reply:
unique_ptr<x> => x is stored on heap
optional<x> => x is stored on stack

Reply:
Well, the optional is stored in this class, so unless this class is stored on the stack, it will just be stored on the heap with the rest of it.

I'm reading bottom to top here, so maybe the stack placement is still coming, but at least right now all SoundSamples in the game get newed on the heap in sound_file_load

Reply:
>  it will just be stored on the heap with the rest of it.

It's about indirection and intent. When it comes to indirection, there is no practical difference here, but I like to avoid it when it isn't necessary.